pkg/images/enricher/enricher_impl.go

```diff
@@ -160,11 +160,7 @@ func (e *enricherImpl) delegateEnrichImage(ctx context.Context, enrichCtx Enrich
 	// or forcing re-scan will trigger updates as necessary.
 	existingImg, exists := e.fetchFromDatabase(ctx, image, enrichCtx.FetchOpt)
 	if exists && cachedImageIsValid(existingImg) {
-		image.Metadata = existingImg.GetMetadata()
-		image.Scan = existingImg.GetScan()
-		image.Signature = existingImg.GetSignature()
-		image.SignatureVerificationData = existingImg.GetSignatureVerificationData()
-		image.Notes = existingImg.GetNotes()
+		e.updateImageWithExistingImage(image, existingImg, enrichCtx.FetchOpt)
```

Comment:
Believe as is the return from `roxctl image scan` would never return a result if a DB entry exists (unless we change handling of `IgnoreExistingImages` mentioned above), or do something like:

```suggestion
		updated := e.updateImageWithExistingImage(image, existingImg, enrichCtx.FetchOpt)
		if updated {
			e.cvesSuppressor.EnrichImageWithSuppressedCVEs(image)
			e.cvesSuppressorV2.EnrichImageWithSuppressedCVEs(image)

			log.Debugf("Delegated enrichment returning cached image for %q", image.GetName().GetFullName())
			return true, nil
		}
```

Reply:
Fixed this.


On the notion of `IgnoreExistingImages` and its usage in the `ScanImage` API:
I feel like the _behavior_ of `IgnoreExistingImages` is IMO fine, where I think the issue lies is that `ScanImage` uses it in the first place. Looking at the history of the code, this seems to have been in place for quite some time (multiple years back).

Within the usage of roxctl, we most of the time advertise to use `roxctl image check`, which does scanning + policy checking. For the detection service API, the default fetch option is `UseCachesIfPossible`. This also totally makes sense, given that for this API a _force_  option is there as well.

I'll file a follow-up PR to change the `ScanImage` API to use `UseCachesIfPossible` as default, the behavior of `IgnoreExistingImages`, specifically the use-case outline with the watched images, makes sense for me, just not for the scanning API to use.